src/utils/createHOCFromMapper.js

```diff
@@ -10,11 +11,16 @@ import { config as obsConfig } from '../setObservableConfig'
 const MAPPERS_INFO = createSymbol('mappersInfo')
 const observablePropType = () => {}
 
+const allMapperComponents = new WeakMap()
+const setMapperComponent = Component => allMapperComponents.set(Component, true)
+export const isMapperComponent = BaseComponent =>
+  typeof BaseComponent === 'function' && allMapperComponents.has(BaseComponent)
+
 const createComponentFromMappers = (mappers, childFactory) => {
   const { observablesKey: OBSERVABLES } = getConfig()
   const CONTEXT_TYPES = { [OBSERVABLES]: observablePropType }
 
-  return class extends Component {
+  const Component = class extends React.Component {
     static [MAPPERS_INFO] = { mappers, childFactory }
```

Comment:
I tried following same pattern as in `createCompactableHOC` (which is what `react-proxy` does) to get rid of `MAPPERS_INFO` and store `{ mappers, childFactory }` as the value in `allMapperComponents`, but was running into errors - don't see the harm in still accessing via `MAPPERS_INFO` as long as we aren't using its presence to determine whether `isMapperComponent()`